batchupload/uploader.py

```diff
@@ -172,36 +184,173 @@ def up_all(in_path, cutoff=None, target='Uploaded', file_exts=None,
         if verbose:
             pywikibot.output(result.get('log'))
 
-        flog.write('%s\n' % result.get('log'))
+        flog.write(result.get('log'))
         os.rename(f, os.path.join(target_dir, base_name))
         os.rename(info_file, os.path.join(target_dir, base_info_name))
         counter += 1
-        flog.flush()
 
-    flog.close()
-    pywikibot.output('Created %s' % logfile)
+    pywikibot.output(flog.close_and_confirm())
+
+
+def up_all_from_url(info_path, cutoff=None, target='upload_logs',
+                    file_exts=None, verbose=False, test=False,
+                    target_site=None):
+    """
+    Upload all images provide as urls in a make_info json file.
+
+    Media (image) files and metadata files with the expected extension .info
+    should be in the same directory. Metadata files should contain the entirety
+    of the desired description page (in wikitext).
+
+    Outputs separate logfiles for files triggering errors, warnings (and
+    successful) so that these can be used in latter runs.
+
+    @param info_path: path to the make_info json file
+    @param cutoff: number of files to upload (defaults to all)
+    @param target: sub-directory for log files (defaults to "upload_logs")
+    @param file_exts: tuple of allowed file extensions (defaults to FILE_EXTS)
+    @param verbose: whether to output confirmation after each upload
+    @param test: set to True to test but not upload (deprecated?)
+    @param target_site: pywikibot.Site to which file should be uploaded,
+        defaults to Commons.
+    """
+    # set defaults unless overridden
+    file_exts = file_exts or FILE_EXTS
+    target_site = target_site or pywikibot.Site('commons', 'commons')
+    target_site.login()
+
+    # load info file
+    info_datas = common.open_and_read_file(info_path, as_json=True)
+
+    # create target directory if it doesn't exist
+    output_dir = os.path.join(os.path.dirname(info_path), target)
+    common.create_dir(output_dir)
+
+    # create all log files
+    logs = {
+        'success': common.LogFile(output_dir, 'success.log'),
+        'warning': common.LogFile(output_dir, 'warnings.log'),
+        'error': common.LogFile(output_dir, 'errors.log'),
+        'general': common.LogFile(output_dir, '¤uploader.log')
```

Comment:
Historical reasons =)

During normal upload there are normally a bunch of logfiles created during pre-processing. All of these were created in the image directory and so I stuck `¤` in front of each to make them all sort together. Now here I have them all go to a separate directory so the prefix shouldn't be necessary.

Removed prefix.